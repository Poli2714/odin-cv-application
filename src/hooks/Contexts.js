import { createContext } from 'react';

const IndexContext = createContext({
  activeIndex: null,
  setActiveIndex: () => {},
});

const KeyContext = createContext({
  activeKey: {
    eduActiveKey: null,
    expActiveKey: null,
  },
  setActiveKey: () => {},
});

const EduErrMsgContext = createContext({
  showEduErrMessage: false,
  setShowEduErrMessage: () => {},
});

const ExpErrMsgContext = createContext({
  showExpErrMessage: false,
  setShowExpErrMessage: () => {},
});

const PersonalInfoContext = createContext({
  personalInfo: {
    profilePicture: '',
    fullName: '',
    currentJobTitle: '',
    email: '',
    mobile: '',
    city: '',
  },
  setPersonalInfo: () => {},
});

const SummaryContext = createContext({
  summary: '',
  setSummary: () => {},
});

const ExperienceContext = createContext({
  experiences: {
    id: {
      jobTitle: '',
      employer: '',
      jobStartDate: '',
      jobEndDate: '',
      jobDescription: '',
    },
  },
  setExperiences: () => {},
});

const EducationContext = createContext({
  educations: {
    id: {
      degree: '',
      university: '',
      eduStartDate: '',
      eduEndDate: '',
      eduInfo: '',
    },
  },
  setEducations: () => {},
});

export {
  EducationContext,
  EduErrMsgContext,
  ExperienceContext,
  ExpErrMsgContext,
  IndexContext,
  KeyContext,
  PersonalInfoContext,
  SummaryContext,
};
