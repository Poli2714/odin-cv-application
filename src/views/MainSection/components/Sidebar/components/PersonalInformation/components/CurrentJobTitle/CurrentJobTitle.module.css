.jobTitle:not(:focus, :placeholder-shown) + .invalid {
  position: relative;

  &::after {
    color: var(--clr-warning);
    content: '⚠️ Not a valid job title';
    font-size: 0.85rem;

    position: absolute;
    bottom: -1.25rem;
    left: 0;
  }
}
