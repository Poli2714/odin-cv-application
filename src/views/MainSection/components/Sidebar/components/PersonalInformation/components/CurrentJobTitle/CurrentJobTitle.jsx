import { useContext } from 'react';

import { InputWrapper } from 'src/components/layout';
import { Label, TextInput } from 'src/components/forms';
import { PersonalInfoContext } from 'src/hooks/Contexts';

function CurrentJobTitle() {
  const { personalInfo, setPersonalInfo } = useContext(PersonalInfoContext);
  const { currentJobTitle } = personalInfo;
  const handleChange = e =>
    setPersonalInfo({ ...personalInfo, currentJobTitle: e.target.value });

  return (
    <InputWrapper>
      <>
        <Label labelName="Job title" labelFor="jobTitle" />
        <TextInput
          name="jobTitle"
          placeholder="React developer"
          value={currentJobTitle}
          onChange={handleChange}
        />
      </>
    </InputWrapper>
  );
}

export default CurrentJobTitle;
