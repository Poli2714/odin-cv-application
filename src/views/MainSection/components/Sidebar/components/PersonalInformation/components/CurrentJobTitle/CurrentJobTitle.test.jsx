import { expect, test, vi } from 'vitest';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { useState } from 'react';

import CurrentJobTitle from './CurrentJobTitle';
import { PersonalInfoContext } from 'src/hooks/Contexts';

vi.mock('src/components/layout', () => ({
  InputWrapper: vi.fn(({ children }) => (
    <div data-testid="input-wrapper">{children}</div>
  )),
}));

vi.mock('src/components/forms', () => ({
  Label: vi.fn(() => <label htmlFor="jobTitle">Test label</label>),
  TextInput: vi.fn(({ value, onChange }) => (
    <input id="jobTitle" name="jobTitle" value={value} onChange={onChange} />
  )),
}));

function MockParentComponent() {
  const [personalInfo, setPersonalInfo] = useState({ currentJobTitle: '' });

  return (
    <PersonalInfoContext.Provider value={{ personalInfo, setPersonalInfo }}>
      <CurrentJobTitle />
    </PersonalInfoContext.Provider>
  );
}

test('renders CurrentJobTitle', () => {
  render(<CurrentJobTitle />);

  expect(screen.getByTestId('input-wrapper')).toMatchSnapshot();
});

test('updates currentJobTitle value as a user types', async () => {
  const user = userEvent.setup();

  render(<MockParentComponent />);
  const input = screen.getByLabelText('Test label');

  expect(input).toHaveValue('');
  await user.type(input, 'Web developer');
  expect(input).toHaveValue('Web developer');
  await user.clear(input);
  await user.type(input, 'hello');
  expect(input).toHaveValue('hello');
});
