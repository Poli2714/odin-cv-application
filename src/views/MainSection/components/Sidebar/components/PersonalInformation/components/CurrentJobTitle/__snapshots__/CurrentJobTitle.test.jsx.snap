// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders CurrentJobTitle 1`] = `
<div
  data-testid="input-wrapper"
>
  <label
    for="jobTitle"
  >
    Test label
  </label>
  <input
    id="jobTitle"
    name="jobTitle"
    value=""
  />
</div>
`;
